gap/attributes/attr.gi

```diff
@@ -1052,6 +1052,188 @@ function(S)
     end;
 end);
 
+InstallMethod(KernelContainment,
+"for two transformations in a semigroup of degree n",
+[IsTransformation, IsTransformation, IsPosInt],
+function(a, b, n)
+    local m, i, idx, q, class;
+    if DegreeOfTransformation(a) > n or
+        DegreeOfTransformation(b) > n then
+        ErrorNoReturn
+          ("Transformation does not belong
+           to a transformation semigroup of degree n");
+    fi;
+    q := [];
+    for class in KernelOfTransformation(a, n) do
+        m := Minimum(class);
+        for i in class do
+            q[i] := m;
+        od;
+    od;
+    for class in KernelOfTransformation(b, n) do
+        idx := q[class[1]];
+        for i in class do
+            if q[i] <> idx then
+                return false;
+            fi;
+        od;
+    od;
+    return true;
+end);
+
+SEMIGROUPS.ExistsTransversal := function(a, b, n)
+    local exists, class, i;
+    if DegreeOfTransformation(a) > n or
+        DegreeOfTransformation(b) > n then
+        ErrorNoReturn("Transformation does not
+         belong to a transformation semigroup of degree n");
+    fi;
+    for class in KernelOfTransformation(a, n) do
+        exists := false;
+        for i in [1 .. Size(class)] do
+            if class[i] ^ a = class[i] ^ b then
+                exists := true;
+                break;
+            fi;
+        od;
+        if exists = false then
+            return false;
+        fi;
+    od;
+    return true;
+end;
+
+InstallMethod(RegularLeqTransformationSemigroup,
+"for a transformation semigroup",
+[IsTransformationSemigroup],
+function(S)
+    local deg;
+    deg := DegreeOfTransformationSemigroup(S);
+    return
+        function(x, y)
+            # Only returns the correct answer if y is a regular element
+            return KernelContainment(x, y, deg)
+             and SEMIGROUPS.ExistsTransversal(x, y, deg);
+        end;
+end);
+
+InstallMethod(MitschLeqSemigroup,
+"for a semigroup",
+[IsSemigroup],
+function(S)
+    if IsInverseSemigroup(S) then
+        return NaturalLeqInverseSemigroup(S);
```

Comment:
```suggestion
        return NaturalLeqInverseSemigroup(S);
     elif IsRegularSemigroup(S) then
       return NambooripadLeqRegularSemigroup(S);
      fi;
```
and as discussed file an issue that `NambooripadLeqRegularSemigroup` is unreasonably slow.